js_modules/dagster-ui/packages/ui-core/src/ticks/EvaluateScheduleDialog.tsx

```diff
@@ -64,132 +76,309 @@ export const EvaluateScheduleDialog = (props: Props) => {
 };
 
 const EvaluateSchedule = ({repoAddress, name, onClose, jobName}: Props) => {
-  const [_selectedTimestamp, setSelectedTimestamp] = useState<{ts: number; label: string}>();
-  const {data} = useQuery<GetScheduleQuery, GetScheduleQueryVariables>(GET_SCHEDULE_QUERY, {
-    variables: {
-      scheduleSelector: {
-        repositoryLocationName: repoAddress.location,
-        repositoryName: repoAddress.name,
-        scheduleName: name,
+  const [selectedTimestamp, setSelectedTimestamp] = useState<{ts: number; label: string}>();
+  const scheduleSelector: ScheduleSelector = useMemo(
+    () => ({
+      repositoryLocationName: repoAddress.location,
+      repositoryName: repoAddress.name,
+      scheduleName: name,
+    }),
+    [repoAddress, name],
+  );
+
+  // query to get the schedule initially
+  const {data: getScheduleData} = useQuery<GetScheduleQuery, GetScheduleQueryVariables>(
+    GET_SCHEDULE_QUERY,
+    {
+      variables: {
+        scheduleSelector,
       },
     },
-  });
+  );
+
+  // mutation to evaluate the schedule
+  const [scheduleDryRunMutation] = useMutation<
+    ScheduleDryRunMutation,
+    ScheduleDryRunMutationVariables
+  >(SCHEDULE_DRY_RUN_MUTATION);
+
+  // mutation to launch all runs
+  const launchMultipleRunsWithTelemetry = useLaunchMultipleRunsWithTelemetry();
+
   const {
     timezone: [userTimezone],
   } = useContext(TimeContext);
   const [isTickSelectionOpen, setIsTickSelectionOpen] = useState<boolean>(false);
   const selectedTimestampRef = useRef<{ts: number; label: string} | null>(null);
   const {viewport, containerProps} = useViewport();
-  const [shouldEvaluate, setShouldEvaluate] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
```

Comment:
you could grab `loading` from the second tuple element that `useMutation` returns on line 100.

Reply:
good find 👀